Scene tools

TheNerdv2 opened the pull request:
Added Healthpacks, Jumpads and associated prefabs. Added prefabs to testscene in a haphazardly manner. Healthpacks don't update between players, if one player takes one, that doesn't happen for another player.

Commit:
Adding in scene objects, initially

Commit:
Added Jump Pad

Added Jump Pad prefab and touched up code

Commit:
Health pack tweaks

Changed it so that health packs don't give health to full health players, changed prefabs, added prefabs to scene

Lonswaya commented on Assets/Scenes/TestScene.unity:
```diff
@@ -77,18 +77,15 @@ LightmapSettings:
     m_PVRDirectSampleCount: 32
     m_PVRSampleCount: 500
     m_PVRBounces: 2
-    m_PVRFilterTypeDirect: 0
```
Please pull the latest version of this scene from master and make your changes from there, this will be very hard to resolve the conflict manually.

Lonswaya commented on Assets/C#/World/HealthPack.cs:
```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
```
Have you tested this across the network, with multiple players? I know we had an issue with players grabbing it at different times, like a player joining a server late.

Lonswaya commented on Assets/C#/World/JumpPad.cs:
```diff
@@ -0,0 +1,27 @@
+public class JumpPad : MonoBehaviour {
+    public int jumpModifier;
+
+    void OnTriggerEnter(Collider col)
```
Consider adding a cooldown to this, on a per-player basis. Imagine if a player accidentally hits it twice with two different hitboxes, they would go flying. Or you could just use the collider layer that the player's capsule is on.

Also this may get terrifying if players ragdoll on this trampoline..

Lonswaya reviewed (changes requested):
Next meeting, please make a build of this and we can all give it a shot. I have a few behaviors that I want to test.

Commit:
Merge branch 'master' into Scene-Tools

Commit:
Changed jumpPads to "BouncePads"

Made it so that jumpPad adds force when stepped on to player, making them into "BouncePads"

Commit:
Merge branch 'master' into Scene-Tools

TheNerdv2 commented:
Updated Jump pad to launch you when you move into it, we should be go for merge.

Commit:
Fixed network syncs for health packs

Commit:
Fixed a bug with healthpacks and server lag

Lonswaya commented:
I made some changes to have the health packs work across networks, so it isnt all clientside. Should work better now.

Later we should take a look at jump pads, because sometimes just standing on them leads to inconsistent results, but it works for now.
